rotkehlchen/chain/ethereum/tokens.py

```diff
@@ -228,25 +315,51 @@ def query_tokens_for_addresses(
         - BadFunctionCallOutput if a local node is used and the contract for the
           token has no code. That means the chain is not synced
         """
-        address_to_balance: Dict[ChecksumEthAddress, Dict[EthereumToken, FVal]] = {}
+        accounts_to_balances: Dict[ChecksumEthAddress, Dict[EthereumToken, FVal]] = {}
         all_tokens = set()
+        accounts_to_tokens: Dict[ChecksumEthAddress, List[EthereumToken]] = {}
+
+        if self.ethereum.connected_to_any_web3():
+            chunk_size = OTHER_MAX_TOKEN_CHUNK_LENGTH
+            # skipping etherscan because chunk size is too big for etherscan
+            call_order = self.ethereum.default_call_order(skip_etherscan=True)
+        else:
+            chunk_size = ETHERSCAN_MAX_ARGUMENTS_TO_CONTRACT
+            call_order = [ETHERSCAN_NODE]
 
         with self.db.conn.read_ctx() as cursor:
             for address in addresses:
                 saved_list, _ = self.db.get_tokens_for_address(cursor, address=address)
                 if saved_list is None:
                     continue  # Do not query if we know the address has no tokens
                 all_tokens.update(saved_list)
+                accounts_to_tokens[address] = saved_list
+
+        regular_chunks = generate_regular_chunks(
+            accounts_to_tokens=accounts_to_tokens,
+            chunk_length=chunk_size,
+        )
+        multicall_chunks = generate_multicall_chunks(
+            accounts_to_tokens=accounts_to_tokens,
+            chunk_length=chunk_size,
+        )
 
+        if len(multicall_chunks) < len(regular_chunks):
```

Comment:
you need to either write more self-documenting code or add a few strategic comments here and there explaining what you are doing. I have no idea what you are doing here or why you are doing it.